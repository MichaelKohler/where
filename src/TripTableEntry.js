import React from 'react';
import './scss/triptable.scss';

class TripTableEntry extends React.Component {
  render() {
    const trip = this.props.trip;

    return (
      <tr>
        <td>{new Date(trip.dateFrom).toDateString()}</td>
        <td>{new Date(trip.dateUntil).toDateString()}</td>
        <td>{trip.destination}</td>
        <td>{trip.country}</td>
        <td>{trip.description}</td>
        <td>{trip.flights}</td>
      </tr>
    );
  }
}

TripTableEntry.defaultProps = {
  trip: {}
};

TripTableEntry.propTypes = {
  trip: React.PropTypes.shape({
    destination: React.PropTypes.string,
    country: React.PropTypes.string,
    dateFrom: React.PropTypes.string,
    dateUntil: React.PropTypes.string,
    description: React.PropTypes.string,
    flights: React.PropTypes.number
  }),
};

export default TripTableEntry;
