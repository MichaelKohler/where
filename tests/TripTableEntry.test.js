import React from 'react';
import { shallow } from 'enzyme';

import TripTableEntry from '../src/TripTableEntry';

test('should correctly shallow render TripTableEntry - past', () => {
  const trip = {
    dateFrom: '2020-02-02',
    dateUntil: '2020-02-03',
    destination: 'Bern',
    country: 'Switzerland',
    description: 'Foo',
  };
  const component = shallow(<TripTableEntry trip={trip} />);
  expect(component).toMatchSnapshot();
});

test('should correctly shallow render TripTableEntry - past - no description', () => {
  const trip = {
    dateFrom: '2020-02-02',
    dateUntil: '2020-02-03',
    destination: 'Bern',
    country: 'Switzerland',
  };
  const component = shallow(<TripTableEntry trip={trip} />);
  expect(component).toMatchSnapshot();
});

test('should correctly shallow render TripTableEntry - future', () => {
  const trip = {
    dateFrom: '2999-02-02',
    dateUntil: '2999-02-03',
    destination: 'Bern',
    country: 'Switzerland',
    description: 'Foo',
  };
  const component = shallow(<TripTableEntry trip={trip} />);
  expect(component).toMatchSnapshot();
});
