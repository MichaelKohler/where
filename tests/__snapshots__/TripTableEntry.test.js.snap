// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly shallow render TripTableEntry - future 1`] = `
<tr
  className={
    Array [
      "future",
    ]
  }
>
  <td>
    Sat Feb 02 2999
  </td>
  <td>
    Sun Feb 03 2999
  </td>
  <td>
    Bern
  </td>
  <td>
    Switzerland
  </td>
  <td>
    Foo
  </td>
</tr>
`;

exports[`should correctly shallow render TripTableEntry - past - no description 1`] = `
<tr
  className={Array []}
>
  <td>
    Sun Feb 02 2020
  </td>
  <td>
    Mon Feb 03 2020
  </td>
  <td>
    Bern
  </td>
  <td>
    Switzerland
  </td>
  <td />
</tr>
`;

exports[`should correctly shallow render TripTableEntry - past 1`] = `
<tr
  className={Array []}
>
  <td>
    Sun Feb 02 2020
  </td>
  <td>
    Mon Feb 03 2020
  </td>
  <td>
    Bern
  </td>
  <td>
    Switzerland
  </td>
  <td>
    Foo
  </td>
</tr>
`;
